test/deploy.smart.wallet.test.ts

```diff
@@ -0,0 +1,61 @@
+import { RelayingServices, SmartWallet } from '../src';
+import { MockRelayingServices, Web3Mock } from './mock';
+import Expect = jest.Expect;
+import {
+    MOCK_SMART_WALLET_ADDRESS,
+    MOCK_TOKEN_ADDRESS,
+    MOCK_TRANSACTION_RECEIPT
+} from './constants';
+
+declare const expect: Expect;
+
+describe('Tests for smart wallet', () => {
+    let sdk: RelayingServices;
+
+    beforeEach(async () => {
+        sdk = new MockRelayingServices();
+    });
+
+    it('Should fail with smart wallet already deployed', async () => {
+        try {
+            await sdk.deploySmartWallet(
+                {
+                    address: MOCK_SMART_WALLET_ADDRESS,
+                    index: 0
+                },
+                MOCK_TOKEN_ADDRESS,
+                0
+            );
```

Comment:
I think we should add a `fail()` in case of the `sdk.deploySmartWallet` call doesn't raise an error

Reply:
This will fixed in the next commit.

Reply:
@wilsoniovlabs could you please explain why in this test case we can expect the smart wallet to be deployed already?

Reply:
Fixed.